### PR TITLE
ROS2/Tinyxml2 updates

### PR DESCRIPTION
**Pull request by pbeeson**
Updates needed for ros2 compile (tested under eloquent) and tinyxml2.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,57 +1,63 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(srdfdom)
 
-find_package(Boost REQUIRED)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
```
Comment (mamoll) [inline CMakeLists.txt:6]: You may also want to set these related flags:
```
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)
```

**Comment by henningkayser** [comment]
@pbeeson Thanks a lot for preparing this! I've created a PR for your feature branch with some changes that should fix CI warnings (https://github.com/jdlangs/srdfdom/pull/1). Please review/merge and then squash the commit history into meaningful chunks (i.e. code migration, update CMakeLists.txt/package.xml, etc...)

**Comment by pbeeson** [comment]
@henningkayser Done.

**Comment by henningkayser** [comment]
@pbeeson thanks for merging. It looks like `urdfdom_py` is still missing. I created a new ROS 2 port of that which should get merged first in order to get CI to succeed. We'll need to add a rosinstall for that.

**Comment by henningkayser** [comment]
@pbeeson thanks

**Pull request merged**